### DISCUSSION TITLE
Assisted pull-ups / chin-ups etc machine

### DISCUSSION THREAD

**Discussion by yrinkel** [astashov/liftosaur#152] [Ideas]
I’m not using those personally, but there was some request about that and since we are collecting ideas anyway. 

Assisted pull-ups / chin-ups and possibly other exercise are using a counter lever to assist people doing those bodyweight exercises with less than their bodyweight. 

Contrary to other machines, the more weight you put, the easier the exercise become. So the progression should be negative, i.e., starting from the user set weight and removing weight as the user progresses. 

It’s easy to program in Liftoscript but for new users, Liftosaur does not currently support native negative progression. 

It could also be nicely used together with the planned ```bodyweight``` variable (https://github.com/astashov/liftosaur/discussions/122#discussion-7089800)

**Comment by galantra** [comment]
> It’s easy to program in Liftoscript

How would you go about doing that?

**Reply by yrinkel** [reply]
Something like this:
This will start with a weight equivalent to your bodyweight (assumed to be 80Kg here) and *decrease* the counterweight every time you complete all sets, making the exercise more and more difficult. At the end, the counterweight reaches 0, meaning you are pulling your whole body and ready to upgrade to normal pull-ups

Pull Up, Leverage Machine / 3x12 / 80kg / progress: custom(plate:5) {~
    if (completedReps >= reps) {
      weights -= state.plate
      if (weights < 0) {
        weights = 0
      }
    }
~}

**Comment by kbrock** [comment]
I always think of this as a negative weight with weighted pull-ups.

Does this make sense to others?

**Reply by galantra** [reply]
Definitely. I've thought that perhaps it is as simple as enabling negative values for the weight variable.

**Comment by richbeales** [comment]
@astashov this can probably be closed due to https://github.com/astashov/liftosaur/commit/55ca42900547dd21741d1f9b844133bbea2b7754

**Comment by ragingdave** [comment]
I think the ability to handle progression by decrementing the weight is good, but how does that factor into 1RM calculations for those exercises. Not that it really applies, but being able to track the eRM where the goal is to decrease the weight doesn't really fit from what I've been seeing.

Sidenote: Love this app, found it this week and can't help but tell everyone I talk to about it because it's absolutely the best out there.

**Comment by astashov** [comment]
You can do it now with this: https://www.reddit.com/r/liftosaur/comments/1n4x690/bodyweight_bar_and_assisted_equipment/
